arroyo-console/src/routes/connections/ConfigureProfile.tsx

```diff
@@ -0,0 +1,254 @@
+import { JSONSchema7 } from 'json-schema';
+import {
+  Button,
+  HStack,
+  Heading,
+  Modal,
+  ModalBody,
+  ModalCloseButton,
+  ModalContent,
+  ModalHeader,
+  ModalOverlay,
+  Spacer,
+  Stack,
+  Text,
+  useDisclosure,
+  IconButton,
+  Box,
+  LinkBox,
+  LinkOverlay,
+  useToast,
+} from '@chakra-ui/react';
+import { useState } from 'react';
+import { JsonForm } from './JsonForm';
+import {
+  AddIcon,
+  ArrowBackIcon,
+  ArrowRightIcon,
+  DeleteIcon,
+  InfoOutlineIcon,
+} from '@chakra-ui/icons';
+import { ConnectionProfile, Connector, del, useConnectionProfiles } from '../../lib/data_fetching';
+import { CreateConnectionState } from './CreateConnection';
+import { formatError } from '../../lib/util';
+import { CreateProfile } from './CreateProfile';
+
+function ClusterViewerModal({
+  isOpen,
+  onClose,
+  profile,
+  connector,
+  schema,
+}: {
+  isOpen: boolean;
+  onClose: () => void;
+  profile: ConnectionProfile;
+  connector: string;
+  schema: JSONSchema7;
+}) {
+  return (
+    <Modal isOpen={isOpen} onClose={onClose} size={'2xl'}>
+      <ModalOverlay />
+      <ModalContent>
+        <ModalHeader>Viewing {connector} connection</ModalHeader>
+        <ModalCloseButton />
+        <ModalBody p={4} px={7}>
+          <JsonForm
+            schema={schema}
+            hasName={true}
+            error={null}
+            initial={profile.config}
+            button={'Close'}
+            buttonColor={'gray'}
+            readonly={true}
+            onSubmit={async () => onClose()}
+          />
+        </ModalBody>
+      </ModalContent>
+    </Modal>
+  );
+}
+
+const ClusterChooser = ({
+  connector,
+  connections,
+  onSubmit,
+  schema,
+  setCreatingCluster,
+}: {
+  connector: string;
+  connections: Array<ConnectionProfile>;
+  onSubmit: (c: string) => void;
+  schema: JSONSchema7;
+  setCreatingCluster: (b: boolean) => void;
+}) => {
+  const [editingProfile, setEditingProfile] = useState<any>({});
```

Comment:
Does this need to be `any`?

Reply:
Ah, not any more; previously this was used for creating profiles as well